### PR TITLE
Enhancement/move loss functions

### PR DESCRIPTION
**Pull request by djbyrne**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Makes the RL implementations a bit more modular to have the specific loss functions as part of the Losses module instead of implemented as part of specific models

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
👍

### CODE REVIEW

File: docs/source/losses.rst [hunk 1]
```diff
@@ -10,3 +10,11 @@ This package lists common losses across research domains
 Your Loss
 ---------
 We're cleaning up many of our losses, but in the meantime, submit a PR to add your loss here!
+
+----------------------------------------
+
+Reinforcement Learning Loss Functions
+----------------------------------------
```
Comment (Borda) [inline docs/source/losses.rst:17]: these shall have the same length as the title

File: docs/source/losses.rst [hunk 2]
```diff
@@ -10,3 +10,11 @@ This package lists common losses across research domains
 Your Loss
 ---------
 We're cleaning up many of our losses, but in the meantime, submit a PR to add your loss here!
+
+----------------------------------------
```
Comment (Borda) [inline docs/source/losses.rst:14]: also, this line shall have a fixed length, pls check either HR in docs

Comment (djbyrne) [reply docs/source/losses.rst]: I am not sure exactly what you mean on this one. Is this not supposed to be used or is there a specific length for seperators? I have made an update, let me know if its correct
